### PR TITLE
Update redis library to support redis v7 (#24114)

### PR DESCRIPTION
**Pull request by pat-s**
#24114

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: modules/cache/cache_redis.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 	"code.gitea.io/gitea/modules/nosql"
 
 	"gitea.com/go-chi/cache"
-	"github.com/go-redis/redis/v8"
+	"github.com/redis/go-redis/v9"
```
Comment (lunny) [inline modules/cache/cache_redis.go:15]: Maybe you mean v9 not v7?

Comment (techknowlogick) [reply modules/cache/cache_redis.go:15]: @lunny v9 of the library supports v7 of redis

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
🤖 🎺

**Comment by pat-s** [comment]
@lunny The GO library is v9 which supports redis-cluster v7.

**Pull request merged**